metagraph/src/annotation/row_diff_builder.cpp

```diff
@@ -451,7 +472,7 @@ void assign_anchors(const std::string &graph_fname,
             optimize_anchors = true;
     }
 
-    sdsl::bit_vector anchors_bv(boss.get_last().size(), false);
+    sdsl::bit_vector anchors_bv(graph.get_last()->size(), false);
```

Comment:
```suggestion
    sdsl::bit_vector anchors_bv(graph.max_index() + 1, false);
```
there's no need to generate a last vector here if we're only using its size

Reply:
@hmusta I merged this, but it seems to trigger the assert in workflows:

https://github.com/ratschlab/metagraph/blob/4f7b681c18d1c05e582d29375eb374e58613c608/metagraph/src/graph/representation/succinct/boss.cpp#L2791

So, the value we put here must _exactly_ match what `graph.get_last()->size()` returns, which is `boss().get_last()` in case of `DBGSuccinct`. I assume in BOSS it's actually not `max_index() + 1`?

Reply:
hmm...maybe I'm missing something in the call tree, but for `DBGSuccinct`, shouldn't it be the case that `max_index() + 1 == num_nodes() + 1 == boss.num_edges() + 1 == boss.get_W().size() - 1 + 1`?

Reply:
Just checked, in that specific test case
```
max_index() + 1: 7,
num_nodes() + 1: 7,
boss.num_edges() + 1: 12,
boss.get_W().size(): 12,
last()->size(): 12
```
Because of dummy nodes maybe?

Reply:
FYI, The test is https://github.com/ratschlab/metagraph/blob/f2f320fa3fd3c9472260abc2abe0857618d8181e/metagraph/tests/annotation/test_aligner_labeled.cpp#L61-L70

Reply:
Ahh, I think I get what's going on now. Some of the unit tests still generate a dummy mask for the BOSS table, so `num_nodes()` skips the dummy edges. In that case, we should use `boss.get_last().size()` for `DBGSuccinct` and `max_index() + 1` for everything else.

Reply:
Also playing around with `kmer_to_boss_index` shows that the mapping is as follows:
```
1 -> (boss) 3
2 -> (boss) 4
3 -> (boss) 6
4 -> (boss) 8
5 -> (boss) 9
6 -> (boss) 10
```

Reply:
Makes sense, the corresponding BOSS edges look like this (with the node IDs after `->` if the graph has a mask)
```
 0 $$$ $
 1 $$$ A
 2 $$$ G
 3 CAA T -> 1
 4 GCA A -> 2
 5 $GC A
 6 TGC T -> 3
 7 $$G C
 8 ATG C -> 4
 9 AAT G -> 5
10 GCT T -> 6
11 CTT $
```